crates/sol-macro-input/src/json.rs

```diff
@@ -27,6 +27,27 @@ impl SolInput {
         let derives =
             attrs.iter().filter(|attr| attr.path().is_ident("derive")).collect::<Vec<_>>();
 
+        // Extract all_derives and extra_derives attributes for propagation to internal contracts
+        let sol_derives = attrs
+            .iter()
+            .filter(|attr| {
```

Comment:
you can extract this logic to a separate fn and merge it above into the derives filter